sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -789,6 +789,11 @@ class kernel_bundle_impl {
   const std::vector<device> &get_devices() const noexcept { return MDevices; }
 
   std::vector<kernel_id> get_kernel_ids() const {
+    // RTC kernel bundles shouldn't have user-facing kernel ids, return an
+    // empty vector when the bundle contains RTC kernels.
+    if (MLanguage == syclex::source_language::sycl_jit) {
```

Comment:
is this necessary? I thought this wasn't returning ids for _any_ language?

from https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc#interaction-with-existing-kernel-bundle-member-functions

> Interaction with existing kernel bundle member functions
> Kernels created from online compilation of source code do not have any associated kernel_id. Therefore, the function kernel_bundle::get_kernel_ids returns an empty vector of kernel_id objects if the kernel bundle was created from a bundle of state bundle_state::ext_oneapi_source.

But if it is necessary, maybe it should be MLanguage set at all, rather than just `sycl_jit`

Reply:
Good point, we can return early here for any `kernel_compiler`-compiled bundle. The `source_language` enum doesn't have a neutral element. In `kernel_bundle_impl`, we check the "being compiled from an `ext_oneapi_source` bundle" as `!MKernelNames.empty()`, so we could expose that check as `kernel_bundle_impl::isBuildFromSource()` or so.